src/EditorFeatures/Core.Wpf/Preview/PreviewFactoryService.cs

```diff
@@ -259,9 +259,9 @@ public Task<DifferenceViewerPreview> CreateAddedDocumentPreviewViewAsync(Documen
             return CreateAddedDocumentPreviewViewAsync(document, DefaultZoomLevel, cancellationToken);
         }
 
-        private Task<DifferenceViewerPreview> CreateAddedDocumentPreviewViewCoreAsync(ITextBuffer newBuffer, PreviewWorkspace workspace, TextDocument document, double zoomLevel, CancellationToken cancellationToken)
+        private async Task<DifferenceViewerPreview> CreateAddedDocumentPreviewViewCoreAsync(ITextBuffer newBuffer, PreviewWorkspace workspace, TextDocument document, double zoomLevel, CancellationToken cancellationToken)
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
do we need to do this here?  if so, can we doc why.  i'd basically like any switch to the main thread to explain why it's necessary.

Reply:
The method uses `IProjectionBufferFactoryService`. In the absence of documentation that this method is free-threaded, we must assume that it can have a UI thread dependency. Since the method is synchronous, it means the method must be called from the UI thread.

vs-threading has analyzers to cover this case, but they are currently disabled in Roslyn.

Reply:
plz comment.

Reply:
done